iroh/src/discovery.rs

```diff
@@ -725,6 +728,21 @@ mod tests {
             .expect("time drift")
             .as_micros() as u64
     }
+
+    #[tokio::test]
+    async fn test_arc_discovery() -> TestResult {
```

Comment:
maybe use something like static_assert instead?

Reply:
I'm unsure what you have in mind.  Could you elaborate a bit more?

Reply:
using https://crates.io/crates/static_assertions you can just check for trait baounds, instead of doing this manual adding

Reply:
I guess that has [`assert_impl_all`](https://docs.rs/static_assertions/1.1.0/static_assertions/macro.assert_impl_all.html) which I could use.  But then, how?  Adding it at build-time would just put it right below the impl... which seems a bit odd, because it's implemented right there.  Plus it would add a new build-time dependency.

I can use it in the test I guess, but again a new dependency seems like a lot.

I added this test mostly to assure myself I did the right thing.  But talking about it here seems to have little value?  Maybe it should just be removed instead?

Reply:
> Maybe it should just be removed instead?

maybe, not sure testing this makes some sense, but also it's a static thing, so I don't think there is much value in the test

Reply:
removed, PTAL